src/components/sprint-view-table/tasks-table.tsx

```diff
@@ -0,0 +1,190 @@
+import { Box, Card, CircularProgress, IconButton } from "@mui/material";
+import { Projects, Tasks, TimeEntries } from "../../generated/homeLambdasClient";
+import { DataGrid } from "@mui/x-data-grid";
+import { useEffect, useState } from "react";
+import { useLambdasApi } from "../../hooks/use-api";
+import { getHoursAndMinutes } from "../../utils/time-utils";
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import strings from "../../localization/strings";
+
+/**
+ * Component properties
+ */
+interface Props {
+  project : Projects,
+  loggedInpersonId?: number,
+  filter?: string
+}
+
+/**
+ * Task component
+ */
+const TaskTable = ({project, loggedInpersonId, filter}: Props) => {
+  const { tasksApi, timeEntriesApi } = useLambdasApi();
+  const [tasks, setTasks] = useState<Tasks[]>([]);
+  const [timeEntries, setTimeEntries] = useState<number[]>([]);
+  const [open, setOpen] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  /**
+   * Gather tasks and time entries when project is available and update changes appeared
+   */
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
As mentioned in the previous screen, move the loading states into the getTasksAndTimeEntries function and then no need to await in the useEffect

Reply:
useEffect(() => {
    if (project && open) {
      getTasksAndTimeEntries();
    }
  }, [project, loggedInpersonId, filter, open]);